File: 백준/Silver/25206. 너의 평점은/너의 평점은.py
res=0
grade=0
for i in range(20):
    a,b,c=map(str,input().split())
    b=float(b)
    if c == "P":
        continue
    grade+=b

    if c == "A+":
        res += 4.5*b
    elif c == "A0":
        res += 4.0*b
    elif c== "B+":
        res += 3.5*b
    elif c== "B0":
        res += 3.0*b
    elif c== "C+":
        res += 2.5*b
    elif c== "C0":
        res += 2.0*b
    elif c== "D+":
        res += 1.5*b
    elif c== "D0":
        res += 1.0*b
    elif c== "F":
        res += 0.0*b

print(res/grade)

File: 백준/Silver/25206. 너의 평점은/너의 평점은.kt
fun main() {

    var sum = 0.0
    var total = 0.0

    repeat(20) {
        val grade = readLine()!!.split(" ")
        val credit = grade[1].toDouble()

        total += credit

        when (grade[2]) {
            "A+" -> sum += (credit * 4.5)
            "A0" -> sum += (credit * 4.0)
            "B+" -> sum += (credit * 3.5)
            "B0" -> sum += (credit * 3.0)
            "C+" -> sum += (credit * 2.5)
            "C0" -> sum += (credit * 2.0)
            "D+" -> sum += (credit * 1.5)
            "D0" -> sum += (credit * 1.0)
            "F" -> sum += (credit * 0.0)
            "P" -> total -= grade[1].toFloat()
        }

    }

    println(sum / total)
}
